### PR TITLE
Add execute context support

### PR DESCRIPTION
**Pull request by brydio**

### CODE REVIEW

**Comment by brydio** [comment]
Add execute context support #150

File: src/pubsub.js [hunk 1]
```diff
@@ -166,7 +166,7 @@
      * @param { Function } func The function to call when a new message is published
      * @return { String }
      */
-    PubSub.subscribe = function( message, func ){
+    PubSub.subscribe = function( message, func, context ){
```
Comment (olleolleolle) [inline src/pubsub.js]: Could you add a documentation annotation for this new parameter (a `@param`)?

File: src/pubsub.js [hunk 2]
```diff
@@ -195,11 +195,11 @@
      * @param { Function } func The function to call when a new message is published
      * @return { PubSub }
      */
-    PubSub.subscribeOnce = function( message, func ){
+    PubSub.subscribeOnce = function( message, func, context ){
```
Comment (olleolleolle) [inline src/pubsub.js]: (And, an annotation is also wanted here - that is: a `@param`.)

**Comment by brydio** [comment]
> Thanks for working to improve PubSubJS! I made tiny notes inline.
> 
> Can this new functionality be tested in unit tests?

ok, i'll update it later

File: src/pubsub.js (hunk 1 above)
Comment (olleolleolle) [reply src/pubsub.js]: Thanks - but a typo snuck in 🐭 !

File: src/pubsub.js (hunk 2 above)
Comment (olleolleolle) [reply src/pubsub.js]: `eontext` => `context`

**Comment by mroderick** [comment]
Have you considered that this can be done natively by ES5 onwards using [`Array.prototype.bind`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind)?

This has been discussed before in https://github.com/mroderick/PubSubJS/issues/21

Is this request any different?

**Pull request closed without merging**